cpp/dolfinx/geometry/BoundingBoxTree.cpp

```diff
@@ -39,24 +39,20 @@ std::array<std::array<double, 3>, 2>
 compute_bbox_of_entity(const mesh::Mesh& mesh, int dim, std::int32_t index)
 {
   // Get the geometrical indices for the mesh entity
-  const int tdim = mesh.topology().dim();
   xtl::span<const double> xg = mesh.geometry().x();
 
-  mesh.topology_mutable().create_connectivity(dim, tdim);
-
   // FIXME: return of small dynamic array is expensive
-  const std::array<std::int32_t, 1> entity = {index};
-  const xt::xtensor<std::int32_t, 2> vertex_indices
+  xtl::span<const std::int32_t> entity(&index, 1);
+  const std::vector<std::int32_t> vertex_indices
       = mesh::entities_to_geometry(mesh, dim, entity, false);
```

Comment:
I believe this isn’t a good idea. Entities to geometry should be an AdjacencyList, as it would have variable width for sub entities of pyramids/prisms

Reply:
> I believe this isn’t a good idea. Entities to geometry should be an AdjacencyList, as it would have variable width for sub entities of pyramids/prisms

There is no functionality change in this PR. Just an improvement in implementation.

Reply:
I think by flattening the data you make it more tedious to go back to an AdjacencyList if we want to support prisms and pyramid facets. However, do you want me to make an issue that can deal with it after this PR?

Reply:
We are still miles from supported pyramid and mixed topologies. No need for hypothetical changes yet.